### ISSUE TITLE
Question : How to use devenv as a replacement for anaconda

### ISSUE THREAD

**Issue by manujchandra** [cachix/devenv#695]
Hi,

I am new to the nix ecosystem. I have a two part question:

First:
I use anaconda as a python distribution. I have created several virtual environments (venv(s)) in it, for eg: quant, util etc. Assume I have used pip to install packages in those venvs.

My use case is that I use packages from the environments without activating them using relative paths as follows:
`~/anaconda3/envs/util/bin/ueberzug`
or 
`~/anaconda3/envs/quant/bin/python ./myscript.py` (myscript.py will be executed in the quant venv because I am specifically using the python from quant venv and will also look for all the imports from quant venv)

How can I do the same with devenv?

I suppose I want to create a file where I declare the pypi packages I want. Then, I want to access the packages without having to activate the environment by using relative paths as in the example given above.

In other words, I want to make my python venvs declarative.

Second:
Once my python venvs declarative are ready and committed to a repo, I want to specify it in the `configuration.nix` file so when I build the system, the venvs also get built along with native Linux packages.

The idea is to have a single (or a set of dependent) configuration files which declare how the OS and multiple venvs are created. So when I build that config, all my linux packages and multiple venvs get created.

I am sure there is a way to do it.

Thanks.

**Comment by domenkozar** [comment]
Hey!

You can create only one virtualenv per developer environment:

```nix
{ pkgs, ... }:

{
  languages.python.enable = true;
  languages.python.version = "3.11";
  languages.python.venv.enable = true;
  languages.python.venv.requirements = ''
    requests
  '';
}
```

Then you run `devenv shell` to activate this environment relative to `devenv.nix`.

You can then create some automation around this logic yourself to be able to switch back and forth between different environments.

**Comment by totoroot** [comment]
As a reference to any other Nix newcomers, it might be helpful to note that you don't have to list all requirements as a multiline string. You can use [`lib.strings.fileContents`](https://ryantm.github.io/nixpkgs/functions/library/strings/#function-library-lib.strings.fileContents) to source the required Python packages directly from your project's `requirements.txt`.

Like so:
```nix
{ lib, ... }:

{
  languages.python = {
    enable = true;
    venv = {
      enable = true;
      requirements = lib.strings.fileContents ./requirements.txt;
    };
  };
```

@domenkozar Do you want me to open a PR to add this note to the documentation?

**Comment by domenkozar** [comment]
I've adopted the [python example](https://github.com/cachix/devenv/blob/main/examples/python/devenv.nix) to this.

I'm not sure how to make this more explicit, maybe we should have per language documentation with descriptions about how to use it.

**Comment by domenkozar** [comment]
@manujchandra can we close this or would you like further clarification?
